CMakeLists.txt

```diff
@@ -145,6 +145,23 @@ if(CLIENTS OR TESTS)
   endif()
 endif()
 
+# Check plugin support
+find_path(DLFCN_H "dlfcn.h")
+if(DLFCN_H)
+  message(STATUS "Found <dlfcn.h>, enabling plugins using POSIX dynamic linker interface")
+  set(PLUGINS TRUE)
+  set(SOURCES_PLUGINS
+    src/utilities/plugin_posix.cpp
+  )
+endif()
+
+if(NOT PLUGINS)
+  message(STATUS "Disabling plugin support")
+  set(SOURCES_PLUGINS
+    src/utilities/plugin_none.cpp
+  )
+endif()
+
```

Comment:
Hmm, yes, #ifdef's would not be nice. I made this remark a bit too early perhaps, before I saw all the changes. If it's not easy then let's not bother.